python/lsst/ts/salobj/sal_info.py

```diff
@@ -766,22 +765,20 @@ async def _read_loop(self, loop: asyncio.AbstractEventLoop) -> None:
 
     def _blocking_read(
         self,
-    ) -> typing.Tuple[
-        typing.Optional[topics.ReadTopic], typing.List[type_hints.BaseMsgType]
+    ) -> typing.List[
+        typing.Tuple[topics.ReadTopic, typing.List[type_hints.BaseDdsDataType]]
     ]:
         """Read DDS data.
 
-        Return two values:
+        Return array of tuples (two values):
```

Comment:
Please update the doc string with new info. Assuming you return an empty list instead of None and assuming you don't return empty data_lists, something like:
```
"""Read DDS data

Returns
-------
read_data
    Read data, as a list of tuples, each with two values:

    * reader : `ReadTopic`
        The read topic.
    * data_list: `list` [`BaseMsgType`]
        List of messages.
"""
```